cmd/config-reloader/Dockerfile

```diff
@@ -1,10 +1,11 @@
-FROM golang:1.17-buster AS buildbase
+FROM --platform=$BUILDPLATFORM golang:1.17-buster AS buildbase
 WORKDIR /app
 COPY . ./
 
 FROM buildbase as appbase
-RUN CGO_ENABLED=0 go build -mod=vendor -o config-reloader cmd/config-reloader/*.go
+ARG TARGETOS TARGETARCH
+RUN GOOS=$TARGETOS GOARCH=$TARGETARCH CGO_ENABLED=0 go build -mod=vendor -o config-reloader cmd/config-reloader/*.go
 
-FROM gcr.io/distroless/static:latest
+FROM --platform=$BUILDPLATFORM gcr.io/distroless/static:latest
```

Comment:
I don't believe this is necessary.

Using `--platform=$BUILDPLATFORM` is useful if you want to cross compile your docker images from x86 without using `docker buildx`.

`docker buildx` will run each of the builds on the target architecture (arm64, etc.) using QEMU.

Because of this, including `--platform=$BUILDPLATFORM` is not required.

Additionally including `TARGETOS` and `TARGETARCH` are also not required for similar reasons. They are used when cross compiling, and since you are using `docker buildx` you're compiling on the native architecture.

Reply:
Thanks for the input @tstapler! I'll take over the PR to make the edits.